### DISCUSSION TITLE
[Enhancement] [dialog] 新增穿透遮罩

### DISCUSSION THREAD

**Discussion by ALI1416** [element-plus/element-plus#17084] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `dialog`

## Description

隐藏遮罩、关闭点击遮罩关闭窗口时，再新增一个选项：穿透遮罩

效果：HelloWorld字符串可以选中

```vue
<template>
  HelloWorld
  <el-button plain @click="dialogVisible = true">
    Click to open the Dialog
  </el-button>
  <el-dialog
      v-model="dialogVisible"
      :close-on-click-modal="false"
      :modal="false"
      draggable
      title="标题"
  >
    Dialog
  </el-dialog>
</template>

<script lang="ts" setup>
import {ref} from 'vue'

const dialogVisible = ref(false)
</script>
```

### Additional comments

_(empty)_

**Comment by zengkaii** [comment]
pointer-events: none;
可以通过这个css属性结合modal-class自己实现

**Reply by ALI1416** [reply]
加到哪里？加到el-dialog标签里，点击事件无效了。有没有示例什么的

**Reply by zengkaii** [reply]
![image](https://github.com/user-attachments/assets/4105c028-a592-4bd6-8b24-468479ae23fc)
![image](https://github.com/user-attachments/assets/d0e7244a-b356-4f22-8d04-e83fb7e2538d)

**Comment by zengkaii** [comment]
[试试这个demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtYnV0dG9uIHBsYWluIEBjbGljaz1cImRpYWxvZ1Zpc2libGUgPSB0cnVlXCI+XG4gICAgQ2xpY2sgdG8gb3BlbiB0aGUgRGlhbG9nXG4gIDwvZWwtYnV0dG9uPlxuXG4gIDxlbC1idXR0b24gcGxhaW4gQGNsaWNrPVwiKCkgPT4ge0VsTWVzc2FnZS5zdWNjZXNzKCdzdWNjZXNzJyl9XCI+XG4gICAgQ2xpY2sgdG8gb3BlbiB0aGUgRGlhbG9nXG4gIDwvZWwtYnV0dG9uPlxuXG4gIDxlbC1kaWFsb2dcbiAgICB2LW1vZGVsPVwiZGlhbG9nVmlzaWJsZVwiXG4gICAgdGl0bGU9XCJUaXBzXCJcbiAgICB3aWR0aD1cIjUwMFwiXG4gICAgbW9kYWwtY2xhc3M9XCJtb2RhbC1jbGFzc1wiXG4gICAgOmJlZm9yZS1jbG9zZT1cImhhbmRsZUNsb3NlXCJcbiAgPlxuICAgIDxzcGFuPlRoaXMgaXMgYSBtZXNzYWdlPC9zcGFuPlxuICAgIDx0ZW1wbGF0ZSAjZm9vdGVyPlxuICAgICAgPGRpdiBjbGFzcz1cImRpYWxvZy1mb290ZXJcIj5cbiAgICAgICAgPGVsLWJ1dHRvbiBAY2xpY2s9XCJkaWFsb2dWaXNpYmxlID0gZmFsc2VcIj5DYW5jZWw8L2VsLWJ1dHRvbj5cbiAgICAgICAgPGVsLWJ1dHRvbiB0eXBlPVwicHJpbWFyeVwiIEBjbGljaz1cImRpYWxvZ1Zpc2libGUgPSBmYWxzZVwiPlxuICAgICAgICAgIENvbmZpcm1cbiAgICAgICAgPC9lbC1idXR0b24+XG4gICAgICA8L2Rpdj5cbiAgICA8L3RlbXBsYXRlPlxuICA8L2VsLWRpYWxvZz5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5pbXBvcnQgeyBFbE1lc3NhZ2UsIEVsTWVzc2FnZUJveCB9IGZyb20gJ2VsZW1lbnQtcGx1cydcblxuY29uc3QgZGlhbG9nVmlzaWJsZSA9IHJlZihmYWxzZSlcblxuY29uc3QgaGFuZGxlQ2xvc2UgPSAoZG9uZTogKCkgPT4gdm9pZCkgPT4ge1xuICBFbE1lc3NhZ2VCb3guY29uZmlybSgnQXJlIHlvdSBzdXJlIHRvIGNsb3NlIHRoaXMgZGlhbG9nPycpXG4gICAgLnRoZW4oKCkgPT4ge1xuICAgICAgZG9uZSgpXG4gICAgfSlcbiAgICAuY2F0Y2goKCkgPT4ge1xuICAgICAgLy8gY2F0Y2ggZXJyb3JcbiAgICB9KVxufVxuPC9zY3JpcHQ+XG48c3R5bGU+XG4ubW9kYWwtY2xhc3N7XG4gIFxuICBwb2ludGVyLWV2ZW50czogbm9uZTtcbn1cbi5lbC1kaWFsb2d7XG5cbiAgcG9pbnRlci1ldmVudHM6IGF1dG87XG59XG48L3N0eWxlPlxuIiwiZWxlbWVudC1wbHVzLmpzIjoiaW1wb3J0IHsgZ2V0Q3VycmVudEluc3RhbmNlIH0gZnJvbSAndnVlJ1xuaW1wb3J0IEVsZW1lbnRQbHVzIGZyb20gJ2VsZW1lbnQtcGx1cydcblxubGV0IGluc3RhbGxlZCA9IGZhbHNlXG5hd2FpdCBsb2FkU3R5bGUoKVxuXG5leHBvcnQgZnVuY3Rpb24gc2V0dXBFbGVtZW50UGx1cygpIHtcbiAgaWYgKGluc3RhbGxlZCkgcmV0dXJuXG4gIGNvbnN0IGluc3RhbmNlID0gZ2V0Q3VycmVudEluc3RhbmNlKClcbiAgaW5zdGFuY2UuYXBwQ29udGV4dC5hcHAudXNlKEVsZW1lbnRQbHVzKVxuICBpbnN0YWxsZWQgPSB0cnVlXG59XG5cbmV4cG9ydCBmdW5jdGlvbiBsb2FkU3R5bGUoKSB7XG4gIGNvbnN0IHN0eWxlcyA9IFsnaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9lbGVtZW50LXBsdXNAbGF0ZXN0L2Rpc3QvaW5kZXguY3NzJywgJ2h0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vZWxlbWVudC1wbHVzQGxhdGVzdC90aGVtZS1jaGFsay9kYXJrL2Nzcy12YXJzLmNzcyddLm1hcCgoc3R5bGUpID0+IHtcbiAgICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgICAgY29uc3QgbGluayA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ2xpbmsnKVxuICAgICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICAgIGxpbmsuaHJlZiA9IHN0eWxlXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2xvYWQnLCByZXNvbHZlKVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdlcnJvcicsIHJlamVjdClcbiAgICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gICAgfSlcbiAgfSlcbiAgcmV0dXJuIFByb21pc2UuYWxsU2V0dGxlZChzdHlsZXMpXG59XG4iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl0sXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwidnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3J1bnRpbWUtZG9tQGxhdGVzdC9kaXN0L3J1bnRpbWUtZG9tLmVzbS1icm93c2VyLmpzXCIsXG4gICAgXCJAdnVlL3NoYXJlZFwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9zaGFyZWRAbGF0ZXN0L2Rpc3Qvc2hhcmVkLmVzbS1idW5kbGVyLmpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0BsYXRlc3QvZGlzdC9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0BsYXRlc3QvXCIsXG4gICAgXCJAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQGVsZW1lbnQtcGx1cy9pY29ucy12dWVAMi9kaXN0L2luZGV4Lm1pbi5qc1wiXG4gIH0sXG4gIFwic2NvcGVzXCI6IHt9XG59IiwiX28iOnt9fQ==)

**Reply by ALI1416** [reply]
非常感谢，这个有效果的。就先凑活用吧
